Sources/SwiftDocC/Infrastructure/Symbol Graph/SymbolGraphRelationshipsBuilder.swift

```diff
@@ -12,11 +12,42 @@ import SymbolKit
 
 /// A set of functions that add relationship information to a topic graph.
 struct SymbolGraphRelationshipsBuilder {
-    /// A namespace for creation of topic-graph related problems.
-    enum NodeProblem {
-        /// Returns a problem about a node with the given precise identifier not being found.
-        static func notFound(_ identifier: String) -> Problem {
-            return Problem(diagnostic: Diagnostic(source: nil, severity: .error, range: nil, identifier: "org.swift.docc.SymbolNodeNotFound", summary: "Symbol with identifier \(identifier.singleQuoted) couldn't be found"), possibleSolutions: [])
+    /// A namespace for creation of symbol-graph-builder related problems.
+    private enum NodeProblem {
+        /// Returns a problem about a source symbol of a given relationship not being found locally.
+        static func sourceNotFound(_ relationship: SymbolGraph.Relationship) -> Problem {
+            return Problem(
+                diagnostic: Diagnostic(
+                    source: nil, severity: .warning, range: nil,identifier: "org.swift.docc.SymbolNodeNotFound",
+                    summary: "Source symbol \(relationship.source.singleQuoted) not found locally, from \(relationship.kind.rawValue.singleQuoted) relationship to \(relationship.target.singleQuoted)",
+                    explanation: """
+                    The "source" of a symbol graph relationship should always refer to a symbol in the same symbol graph file.
```

Comment:
One thing we might want to consider here is direct the user to file a bug report on Swift/Clang with information that can help diagnose the issue (in our case, the symbol graph file or at least a portion of it).

Reply:
Yes, I was thinking of that but at the point where we're creating these diagnostics we no longer know which symbol graph file a given relationship came from. 

We could include information about how to report an issue for both Swift and Clang and let the developer decide which to file a bug report on.